ansible/roles/python/templates/etc/ansible/facts.d/python.fact.j2

```diff
@@ -29,4 +31,29 @@ output = loads("""{{ ({'enabled':  python__enabled | bool,
 output['installed2'] = cmd_exists('python2')
 output['installed3'] = cmd_exists('python3')
 
+if output['installed2']:
+    try:
+        output['version2'] = subprocess.check_output([
+            'python2', '-c',
+            textwrap.dedent("""
+            from __future__ import print_function
+            import platform
+            print(platform.python_version(), end="")
+            """)
+        ])
+    except subprocess.CalledProcessError:
+        pass
+
+if output['installed3']:
+    try:
+        output['version3'] = subprocess.check_output([
+            'python3', '-c',
+            textwrap.dedent("""
+            import platform
+            print(platform.python_version(), end="")
+            """)
+        ])
+    except subprocess.CalledProcessError:
+        pass
+
```

Comment:
It's a good idea to add the available Python version to the facts, however it's kind of redundand since Ansible does this in its "internal" facts as well, although only for the interpreter. However, why did you go with such an elaborate way to do this? Why not just go with parsing the output of `python3 --version` and `python2 --version` instead? I'm curious because this script will be executed frequently during Ansible runs, so I'm not sure which method would be faster.

Reply:
Because `platform.python_version` is the API driven way to do it and I expect it to be more stable/reliable. I did some testing and it does not make much of a difference but I changed it anyway.

One of those version checks could be switched to Ansible "internal" facts. But not both. For my use case, it did not work obviously thus I added it to the local facts. I still have Python2 as `interpreter_python` but I want to run EteSync (for which this Python version is for) under Python3.